qa/workunits/rbd/rbd-ggate.sh

```diff
@@ -142,9 +161,11 @@ rbd info ${POOL}/${IMAGE}
 if [ -z "$RBD_GGATE_RESIZE_SUPPORTED" ]; then
     # XXX: ggate device resize is not supported by vanila kernel.
     # rbd-ggate should terminate when detecting resize.
-    _sudo rbd-ggate list | expect_false grep " ${DEV} *$"
+    # But that is not consistant for the time ignore
+    rbd-ggate list | expect_false grep " ${DEV} *$" ||  \
+	_sudo rbd-ggate unmap ${devname}
```

Comment:
Could you please provide more details about this? What kernel version do you observe this on?
Note, recently I have committed ggate resize support to freebsd current [1]. So might you just run a kernel that supports resize? Then you just need to set `RBD_GGATE_RESIZE_SUPPORTED` env variable. I added this variable because early resize would work only for my local custom kernel with resize patch applied. Now that it is in vanila kernel probably it makes sense to update this to check for kernel version (via `uname` or `freebsd-version -k`) instead of relying on `RBD_GGATE_RESIZE_SUPPORTED` (and update the comment accordingly). I still don't see much sense in adding yet another comment about this.

[1] https://github.com/freebsd/freebsd/commit/414774b7931d98c2bc5f42317ea69c8a32eebc6e

Reply:
I'm running my Jenkins builds on FreeBSD 12.0-ALPHA2 #42 r329756:337961M of last friday. 
I tried setting `RBD_GGATE_RESIZE_SUPPORTED` on `uname -r | grep -e "^12" ` but that was not very succesful. The ggate-dev stayed, but did not resize. 
I also have a testcluster running on 11.2, and there it is a lot easier to manually play with things, but there it did not work (as expected)
There the ggate-device reliably disappeared when the image was resized.

I added the comment mainly because I did the tests and it does/did not really fully worked as expected. But it is throwing of my Jenkins testing.
Perhaps we can test on the svn sequence of your commit and enable for svn-seq after that
Or look at `uname -K` which now is `1200078`
I will test and update accordingly.

Reply:
@trociny 
That seems to work in the new kernel
```
2: + uname -K
2: + [ 1200078 -ge 1200078 ]
2: + RBD_GGATE_RESIZE_SUPPORTED=1

2: resize test
2: + basename /dev/ggate0
2: + devname=ggate0
2: + geom gate list ggate0
2: + awk '$1 ~ /Mediasize:/ {print $2}'
2: + size=67108864
2: + test -n 67108864
2: + rbd resize testrbdggate93517/test --size 128M
2: 2018-08-19 13:49:59.126 80d5ab000 -1 WARNING: all dangerous and experimental features are enabled.
2: 2018-08-19 13:49:59.126 80d5ab000 -1 WARNING: all dangerous and experimental features are enabled.
2: 2018-08-19 13:49:59.136 80d5ab000 -1 WARNING: all dangerous and experimental features are enabled.
2: ^MResizing image: 100% complete...done.
2: + rbd info testrbdggate93517/test
2: 2018-08-19 13:50:00.616 80d5ab000 -1 WARNING: all dangerous and experimental features are enabled.
2: 2018-08-19 13:50:00.616 80d5ab000 -1 WARNING: all dangerous and experimental features are enabled.
2: 2018-08-19 13:50:00.626 80d5ab000 -1 WARNING: all dangerous and experimental features are enabled.
2: rbd image 'test':
2:      size 128 MiB in 32 objects
2:      order 22 (4 MiB objects)
2:      snapshot_count: 0
2:      id: 1027834d
2:      block_name_prefix: rbd_data.1027834d
2:      format: 2
2:      features: layering, exclusive-lock, object-map, fast-diff, deep-flatten
2:      op_features:
2:      flags:
2:      create_timestamp: Sun Aug 19 13:48:18 2018
2: + [ -z 1 ]
2: + rbd-ggate list
2: + grep ' /dev/ggate0 *$'
2: 0  testrbdggate93517 test  -    /dev/ggate0
2: + geom gate list ggate0
2: + awk '$1 ~ /Mediasize:/ {print $2}'
2: + size2=134217728
2: + test -n 134217728
2: + test 134217728 -eq 134217728
2: + dd 'if=/dev/urandom' 'of=/tmp/tmp.0irvU2Ns/data' 'bs=1M' 'count=128'
2: 128+0 records in
2: 128+0 records out
2: 134217728 bytes transferred in 4.126440 secs (32526275 bytes/sec)
2: + _sudo dd 'if=/tmp/tmp.0irvU2Ns/data' 'of=/dev/ggate0' 'bs=1M'
```

Reply:
@trociny 
For completeness SVN commit is:
https://svnweb.freebsd.org/base?view=revision&revision=336237

Reply:
@trociny 
I would still like to keep the code to remove a leftover dev-ggate in case of no-resize.
But I'll run master on 11.2 to see wat it brings.

Reply:
The idea is that it is not expected to leave a leftover on OSes that does not support resize. If it is not the case in your tests then I would rather investigate this and probably fix. Adding `|| unmap` will just mask a potential issue.